File: BaseLua/src/main/java/org/anime_game_servers/lua/models/IntLuaValue.java
package org.anime_game_servers.lua.models;

public class IntLuaValue extends MockLuaValue {
    public static final IntLuaValue ZERO = new IntLuaValue(0);
    public static final IntLuaValue ONE = new IntLuaValue(1);
    public static final IntLuaValue N_ONE = new IntLuaValue(-1);


    private final int value;

    public IntLuaValue(int value) {
        this.value = value;
    }

    @Override
    public boolean isInteger() {
        return true;
    }

    @Override
    public boolean asBoolean() {
        return value != 0;
    }

    @Override
    public int asInteger() {
        return value;
    }

    @Override
    public long asLong() {
        return value;
    }

    @Override
    public double asDouble() {
        return value;
    }

    @Override
    public float asFloat() {
        return value;
    }

    @Override
    public String asString() {
        return Integer.toString(value);
    }

    @Override
    public <T> T asObject(Class<T> type) {
        if(type == Number.class)
            return type.cast(value);
        return null;
    }
}

File: BaseLua/src/main/java/org/anime_game_servers/lua/models/IntLuaValue.kt
package org.anime_game_servers.lua.models

class IntLuaValue(private val value: Int) : MockLuaValue() {
    override fun isInteger() = true

    override fun asBoolean() = value != 0
    override fun asInteger() = value
    override fun asLong() = value.toLong()
    override fun asDouble() = value.toDouble()
    override fun asFloat() = value.toFloat()
    override fun asString() = value.toString()

    override fun <T> asObject(type: Class<T>): T? {
        return if (type == Number::class.java) type.cast(value) as T? else null
    }

    companion object {
        @JvmField
        val ZERO = IntLuaValue(0)
        @JvmField
        val ONE = IntLuaValue(1)
        @JvmField
        val N_ONE = IntLuaValue(-1)
    }
}
